src/status.ts

```diff
@@ -0,0 +1,267 @@
+          status.busy = true;
+          addGem()
+            .then(() => status.dispose())
+            .catch(() => {});
+        })
+      );
+
+      status.command = {
+        title: "Run bundle add and install",
+        command: "ruby-lsp.addMissingGem",
+      };
+      return true;
+    }
+    return false;
+  }
+
+  private async activateGemOutdatedButton() {
+    const gemOutdated = await isGemOutdated();
+    if (!gemOutdated) {
+      return;
+    }
+
+    const commandId = "updateOutdatedGem";
+
+    const status: vscode.LanguageStatusItem = this.createStatusItem(
+      "updateGem",
+      "Ruby LSP: The gem is not up-to-date",
+      vscode.LanguageStatusSeverity?.Warning
+    );
+
+    this.context.subscriptions.push(
+      vscode.commands.registerCommand(commandId, async () => {
+        status.text = "Ruby LSP: Updating Ruby LSP";
+        status.busy = true;
+        status.command = undefined;
+
+        const result = await updateGem();
+
+        if (result.stderr.length > 0) {
+          status.text = "Ruby LSP: Update failed";
+          status.command = {
+            title: "Try again",
+            command: commandId,
+          };
+
+          if (
+            result.stderr.includes(
+              "Bundler attempted to update ruby-lsp but its version stayed the same"
+            )
+          ) {
+            vscode.window.showWarningMessage(
+              "Could not update the ruby-lsp gem. Is the version in the Gemfile pinned?"
+            );
+          } else {
+            vscode.window.showErrorMessage("Failed to update gem.");
+          }
+        } else {
+          status.dispose();
+        }
+      })
+    );
+
+    status.command = {
+      title: "Update",
+      command: commandId,
+    };
+  }
+
+  private createYjitStatus(ruby: Ruby) {
+    const useYjit = vscode.workspace.getConfiguration("rubyLsp").get("yjit");
+
+    let [major, minor]: number[] = [0, 0];
+
+    if (this.ruby.rubyVersion) {
+      [major, minor] = this.ruby.rubyVersion.split(".").map(Number);
+    }
+
+    const yjit: vscode.LanguageStatusItem =
+      vscode.languages.createLanguageStatusItem("yjit", this.selector);
+    yjit.name = "Ruby LSP Status";
+
+    if (useYjit && ruby.yjitEnabled && [major, minor] >= [3, 2]) {
+      yjit.text = "Ruby LSP: YJIT is in use";
+    } else {
+      yjit.text = "Ruby LSP: YJIT is not in use";
+      if ([major, minor] >= [3, 2]) {
+        yjit.severity = vscode.LanguageStatusSeverity?.Warning;
+        yjit.command = {
+          title: "Enable it",
+          command: "",
+        };
+      } else {
+        yjit.severity = vscode.LanguageStatusSeverity?.Information;
+      }
+    }
+  }
+
+  private createRubyStatus(ruby: Ruby) {
+    const rubyVersion: vscode.LanguageStatusItem =
+      vscode.languages.createLanguageStatusItem("rubyVersion", this.selector);
+    rubyVersion.name = "Ruby LSP Status";
+    rubyVersion.text = `Using Ruby ${ruby.rubyVersion}`;
+  }
```

Comment:
I included this because it's part of the issue, but the Ruby version is already in the status bar (at least for me). Does anyone else already have the Ruby version in their status bar (left side, all the way to the right)? I don't know where this is coming from because I checked `vscode-ruby-shopify` and `sorbet-vscode` and from quickly glancing over that, I couldn't find this in the source code. If this is something most people already have visible, is this still necessary?

Reply:
I also have the Ruby version in my status bar.

Reply:
LOL turns out it's in the one place I didn't check: this project. It's in `ruby.ts`. I'm removing it from there so now the Ruby version is only in one place in the status bar as a status item